plugin/storage/cassandra/dependencystore/storage.go

```diff
@@ -26,60 +26,110 @@ import (
 	casMetrics "github.com/jaegertracing/jaeger/pkg/cassandra/metrics"
 )
 
+// IndexMode determines how the dependency data is indexed.
+type IndexMode int
+
+// IsValid returns true if the IndexMode is a valid one.
+func (i IndexMode) IsValid() bool {
+	return i < end
+}
+
 const (
-	depsInsertStmt = "INSERT INTO dependencies(ts, ts_index, dependencies) VALUES (?, ?, ?)"
-	depsSelectStmt = "SELECT ts, dependencies FROM dependencies WHERE ts_index >= ? AND ts_index < ?"
+	// SASIEnabled is used when the dependency table is SASI indexed.
+	SASIEnabled IndexMode = iota
```

Comment:
would it not make sense to simply refer to this as `v2` throughout, including constants, variables, the CLI flags? SASI then becomes a side effect that the user doesn't really need to know about.